workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/SendCreatePlayerRequestSystem.cs

```diff
@@ -29,48 +42,116 @@ protected override void OnCreateManager()
             commandSystem = World.GetExistingManager<CommandSystem>();
             logDispatcher = workerSystem.LogDispatcher;
 
-            initializationGroup = GetComponentGroup(
-                ComponentType.ReadOnly<WorkerEntityTag>(),
-                ComponentType.ReadOnly<OnConnected>()
-            );
+            QueryForPlayerCreators();
+
+            if (PlayerLifecycleConfig.AutoRequestPlayerCreation)
+            {
+                RequestPlayerCreation();
+            }
+        }
+
+        private void QueryForPlayerCreators()
+        {
+            playerCreatorQueryId = commandSystem.SendCommand(new WorldCommands.EntityQuery.Request
+            {
+                EntityQuery = playerCreatorQuery
+            });
+            ++playerCreatorQueryAttempts;
         }
 
         public void RequestPlayerCreation(byte[] serializedArguments = null)
```

Comment:
If a user calls this inbetween a retry attempt - we end up overriding the parameters? 

user says: send request with params A
request A1 sent
request A1 fails 
request A2 sent
user says: send request with params B
request A2 fails
request B1 sent
--- request A lost to the void ---

Is that acceptable? 

Should we allow users to call this multiple times? Should we retry request A N times until final failure and then do request B or do they run concurrently? 

Not sure on the use cases here

Reply:
I've updated it now so that once a request with args A has been requested, any further requests (ie B in your example) will not be considered until A succeeds, fails concretely or finishes retrying.

I reckon doing this + giving the option to manually enable/disable retries + perhaps giving a callback option for success/failure is probably more flexible and better for the user